### PR TITLE
IA-650 react table upgrade

### PR DESCRIPTION
**Pull request by beygorghor**
- upgrade to latest react-table release
- rewrite and split table component on blsq-component:
       Custom pagination, expand collapse, ...
-  fix issues with actual code base
- fix failing tests
- change react-quill to react-quilljs (more recent)
- remove unused css

⚠️ using thins branch for now https://github.com/BLSQ/bluesquare-components/pull/6

https://user-images.githubusercontent.com/12494624/130614076-16ea9423-f524-4519-a60a-7d632fa3e83b.mov

Screenshot 2021-08-24 at 10 52 49
Screenshot 2021-08-24 at 09 15 00

### CODE REVIEW

File: hat/assets/js/apps/Iaso/index.js [hunk 1]
```diff
@@ -12,7 +12,13 @@ import { store, history } from './redux/store';
...
 import { getPlugins, PluginsContext } from './utils';
 
-const queryClient = new QueryClient();
+const queryClient = new QueryClient({
+    defaultOptions: {
+        queries: {
+            refetchOnWindowFocus: false,
+        },
+    },
+});
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/index.js:21]: not really part of the table stuff but I'm all for it

File: plugins/polio/js/src/components/Dashboard.js [hunk 2]
```diff
@@ -3,11 +3,10 @@ import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import {
     Table,
     textPlaceholder,
-    IconButton as IconButtonComponent,
-    ColumnText,
     LoadingSpinner,
+    IconButton as IconButtonComponent,
 } from 'bluesquare-components';
-import 'react-table/react-table.css';
+// import 'react-table/react-table.css';
```
Comment (olethanh) [inline plugins/polio/js/src/components/Dashboard.js:9]: remove comment

**Review by quang-le — CHANGES_REQUESTED** [review]
Nice job, I like the new look of the table (I think the little shadow effect makes it classier than the previous one).
Code review is in the comments. Nothing blocking, except maybe the condition in `useGetPages`

I tested the app, here's what I found:

- You need to click twice for the sorting happen, but only in one direction. It seems the call is not made on the first click:

https://user-images.githubusercontent.com/38907762/130796080-f9a971cf-3122-4508-bc72-5063213f251f.mov

- In Polio and Pages, sorting will cause the page to flicker:

polio

https://user-images.githubusercontent.com/38907762/130796307-c65ed9b3-3cb9-44b7-a590-027eb3f878b2.mov

pages


https://user-images.githubusercontent.com/38907762/130796428-07bbade4-57af-49d2-bd41-4054c6b52048.mov


- History screen can crash because of a null error:


https://user-images.githubusercontent.com/38907762/130796659-3e948596-b2d6-417f-9086-1ada0b23dc92.mov


- In History screen, when sorting while the row is expanded, the expanded row remains the same, while the table sorting order changes:


https://user-images.githubusercontent.com/38907762/130797111-1399c35d-59de-44ad-a291-74ac92bb63a3.mov


-  When sorting Tables with lots of data, clicking on the column header multiple times will create as many calls, and as many refreshes of the Table. The previous version had a loading screen which hid this to the user. I think we should either reintroduce a loading screen, or disabling sorting while the awaiting the request's response:


https://user-images.githubusercontent.com/38907762/130797739-9fa50646-f75a-41f3-9c5b-00a68feaa808.mov


- In the SideBar, the link to Archived (forms) redirects to List:


https://user-images.githubusercontent.com/38907762/130798104-f17e147a-415c-4acb-9f14-b6aa348755a3.mov


- The Algorithms page is broken:

https://user-images.githubusercontent.com/38907762/130798284-427258d7-aded-460c-9c81-7e13796f582d.mov


- When resizing, the cursor doesn't stick to the header, making it difficult to control with large sizes:


https://user-images.githubusercontent.com/38907762/130798584-06b28380-7e1d-438a-98be-d9dd3b2a61be.mov



https://user-images.githubusercontent.com/38907762/130798624-f70e0f7d-fc5a-49e6-897e-df958ead8744.mov


- The cursor changes to text when hovering over a non-sortable table header. Previous behaviour was to use default cursor:


https://user-images.githubusercontent.com/38907762/130799893-96e45c69-2648-443f-8e80-a9424e44a074.mov


- The tooltip for table headers is not styled like other tooltips (eg: action buttons). The texxt could also be changed to something like "Toggle sort order" or "Change sort order":

Screenshot 2021-08-25 at 12 24 08


- The pagination doesn't have enough room to display page counts over 99:

Screenshot 2021-08-25 at 13 58 48


Sorry, lots of nitpicking because so many parts of the app are impacted. I can't wait to use it though!

Great job!

File: hat/assets/js/test/utils/redux.js [hunk 3]
```diff
@@ -56,7 +58,7 @@ const initialState = {
 export const renderWithStore = (component, state = null) => (
     <Provider store={getMockedStore({ ...initialState, ...state })}>
         <LinkProvider linkComponent={Link}>
-            {renderWithIntl(component)}
+            {renderWithMuiTheme(renderWithIntl(component))}
         </LinkProvider>
```
[the remark this answers is not in the document]
Comment (beygorghor) [reply hat/assets/js/test/utils/redux.js:62]: Yes , create a ticket please

File: hat/assets/js/apps/Iaso/components/filters/chips/ChipsFilterComponent.js [hunk 4]
```diff
@@ -112,7 +112,7 @@ class ChipsFilterComponent extends Component {
                     ))}
                 {notSelectedItems.length > 0 && (
                     <InputComponent
-                        keyValue="form_id"
+                        keyValue={`${locationsKey}-not-selected`}
                         onChange={(key, formId) => this.onAdd(formId)}
```
[the remark this answers is not in the document]
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/components/filters/chips/ChipsFilterComponent.js:116]: yep

File: hat/assets/js/apps/Iaso/domains/completeness/messages.js [hunk 5]
```diff
@@ -33,7 +33,7 @@ const MESSAGES = defineMessages({
         defaultMessage: 'June',
         id: 'iaso.label.months.june',
     },
-    july: {
+    jully: {
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/domains/completeness/messages.js:36]: err?

**Review by olethanh — CHANGES_REQUESTED** [review]

**Pull request merged**